applications/RANSModellingApplication/custom_strategies/generic_residualbased_simple_steady_scalar_scheme.h

```diff
@@ -0,0 +1,377 @@
+                                                LocalSystemMatrixType& LHS_Contribution,
+                                                LocalSystemVectorType& RHS_Contribution,
+                                                Condition::EquationIdVectorType& EquationId,
+                                                ProcessInfo& CurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        const GeometryType& r_geometry = rCurrentCondition->GetGeometry();
+        const double local_delta_time = this->GetGeometryLocalTimeStep(r_geometry);
+
+        rCurrentCondition->SetValue(DELTA_TIME, local_delta_time);
+
+        rCurrentCondition->InitializeNonLinearIteration(CurrentProcessInfo);
+        rCurrentCondition->CalculateLocalSystem(
+            LHS_Contribution, RHS_Contribution, CurrentProcessInfo);
+
+        Matrix SteadyLHS;
+        rCurrentCondition->CalculateLocalVelocityContribution(
+            SteadyLHS, RHS_Contribution, CurrentProcessInfo);
+        rCurrentCondition->EquationIdVector(EquationId, CurrentProcessInfo);
+
+        if (SteadyLHS.size1() != 0)
+            noalias(LHS_Contribution) += SteadyLHS;
+
+        // AddRelaxation(rCurrentCondition->GetGeometry(), local_delta_time,
+        //               LHS_Contribution, RHS_Contribution, CurrentProcessInfo);
+
+        KRATOS_CATCH("");
+    }
+
+    void Calculate_RHS_Contribution(Element::Pointer rCurrentElement,
+                                    LocalSystemVectorType& rRHS_Contribution,
+                                    Element::EquationIdVectorType& rEquationId,
+                                    ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        Matrix LHS_Contribution;
+        CalculateSystemContributions(rCurrentElement, LHS_Contribution, rRHS_Contribution,
+                                     rEquationId, rCurrentProcessInfo);
+
+        KRATOS_CATCH("");
+    }
+
+    void Condition_Calculate_RHS_Contribution(Condition::Pointer rCurrentCondition,
+                                              LocalSystemVectorType& rRHS_Contribution,
+                                              Element::EquationIdVectorType& rEquationId,
+                                              ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        Matrix LHS_Contribution;
+        Condition_CalculateSystemContributions(rCurrentCondition, LHS_Contribution,
+                                               rRHS_Contribution, rEquationId,
+                                               rCurrentProcessInfo);
+
+        KRATOS_CATCH("");
+    }
+    void FinalizeNonLinIteration(ModelPart& rModelPart,
+                                   TSystemMatrixType& rA,
+                                   TSystemVectorType& rDx,
+                                   TSystemVectorType& rb) override
+    {
+        // rModelPart.GetProcessInfo()[STEP] += 1;
+        // mVtkOutput->PrintOutput();
```

Comment:
If a function is just empty, it would probably make sense to not reimplement (and just fall back to the base class)